doc/tests/optimization_functions.tex

```diff
@@ -117,7 +125,43 @@ \subsubsection{ND Slant}
 This trivial linear problem extends to arbitrary dimension and has optimal points at infinities,
 which is usually outside the domain of exploration.
 \begin{itemize}
-  \item Function: $f(\vec x) = 1 - \frac{1}{N}\sum_{x_i\in \vec x} x_i$
+  \item Objective Function: $f(\vec x) = 1 - \frac{1}{N}\sum_{x_i\in \vec x} x_i$
   \item Domain: $0 \leq x_i \leq 1 \forall x_i \in \vec x$
   \item Global Minimum: $f(\{1\}_N) = 0$
 \end{itemize}
+
+\subsubsection{Rosenbrock with cubic and linear constraints}
+Here the Rosenbrock is another two dimensional highly nonlinear function. The global maximum is on the intersection of the two constraints which renders this problem challenging for optimizers that uses random decisions for next points such as simulated annealing and genetic algorithms.
+\begin{itemize}
+	\item Objective Function: $f(x,y) = (1-x^2) + 100(y-x^2)^2$
+	\item Domain: $-1.5 \leq x \leq 1.5$ and $ -0.5 \leq y \leq 2.5$
+	\item Constraints: $(x-1)^3 - y +1 \leq 0$ and $x+y-2 \leq 0$
+	\item Global Maximum: $f(1.0,1.0) = 0$
+\end{itemize}
+
+\subsubsection{Rosenbrock with a disk constraint}
+This is the same 2D function, but constrained to a disk. This constraint makes it a little easier than the previous function.
+\begin{itemize}
+	\item Objective Function: $f(x,y) = (1-x^2) + 100(y-x^2)^2$
+	\item Domain: $-1.5 \leq x,y \leq 1.5$
+	\item Constraint: $x^2 + y^2 \leq 2$
+	\item Global Maximum: $f(1.0,1.0) = 0$
+\end{itemize}
+
+\subsubsection{Townsend}
+It is a highly nonlinear non convex function with multiple minima and maxima. The constraint is highly nonlinear and the y-domain is asymmetric.
+\begin{itemize}
+	\item Objective Function: $f(x,y) = -\left[\cos((x-0.1)y)\right]^2-x\sin(3x+y)$
+	\item Domain: $-2.25 \leq x \leq 2.25$ and $-2.5 \leq y \leq 1.75$
+	\item Constraint: $x^2 + y^2 \leq \left[2\cos(t)-\frac{1}{2}\cos(2t)-\frac{1}{4}\cos(3t) - \frac{1}{8}cos(4t) \right]^2+\left[2\sin(t)\right]^2 $;  \[ \text{where: } t=\arctan2(x,y) \]
+	\item Global Minimum: $f(2.0052938,1.1944509) = -2.0239884$
+\end{itemize}
+
+\subsubsection{Simionescu}
```

Comment:
Do you have all the tests that corresponding to the objective functions that you added?

Reply:
I think one of them took forever to converge so I removed it. Also, I still need to add the discrete ones, and Diego has two or three that still need to be added.

Reply:
If it is not used in the tests, I would suggest to comment it out in the document for now. @Jimmy-INL